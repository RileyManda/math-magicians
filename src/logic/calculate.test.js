import calculate from './calculate';

describe('calculate', () => {
  it('should update calculator data object with AC button pressed', () => {
    const buttonName = 'AC';

    const initialData = {
      total: '10',
      next: '5',
      operation: '+',
    };

    calculate(initialData, buttonName);

    expect(buttonName).toEqual('AC');
  });
  it('should update calculator data object with 1 button pressed', () => {
    const buttonName = '1';

    const initialData = {
      total: '10',
      next: '5',
      operation: '+',
    };

    calculate(initialData, buttonName);

    expect(buttonName).toEqual('1');
  });
});
