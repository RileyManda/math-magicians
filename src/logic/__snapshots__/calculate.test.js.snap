// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate jest snapshots should update calculator data object when % button is pressed 1`] = `
Object {
  "next": null,
  "operation": "%",
  "total": "15",
}
`;

exports[`calculate jest snapshots should update calculator data object when + button is pressed 1`] = `
Object {
  "next": null,
  "operation": "+",
  "total": "15",
}
`;

exports[`calculate jest snapshots should update calculator data object when +/- button is pressed 1`] = `
Object {
  "next": "-5",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when . button is pressed 1`] = `
Object {
  "next": "5.",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when = button is pressed 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "15",
}
`;

exports[`calculate jest snapshots should update calculator data object when - button is pressed 1`] = `
Object {
  "next": null,
  "operation": "-",
  "total": "15",
}
`;

exports[`calculate jest snapshots should update calculator data object when 0 button is pressed 1`] = `
Object {
  "next": "50",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 1 button is pressed 1`] = `
Object {
  "next": "51",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 2 button is pressed 1`] = `
Object {
  "next": "52",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 3 button is pressed 1`] = `
Object {
  "next": "53",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 4 button is pressed 1`] = `
Object {
  "next": "54",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 5 button is pressed 1`] = `
Object {
  "next": "55",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 6 button is pressed 1`] = `
Object {
  "next": "56",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 7 button is pressed 1`] = `
Object {
  "next": "57",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 8 button is pressed 1`] = `
Object {
  "next": "58",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when 9 button is pressed 1`] = `
Object {
  "next": "59",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate jest snapshots should update calculator data object when AC button is pressed 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;
